pygls/feature_manager.py

```diff
@@ -49,7 +49,7 @@ def has_ls_param_or_annotation(f, annotation):
     try:
         sig = inspect.signature(f)
         first_p = next(itertools.islice(sig.parameters.values(), 0, 1))
-        return first_p.name == PARAM_LS or first_p.annotation is annotation
+        return first_p.name == PARAM_LS or get_type_hints(f)[first_p.name] == annotation
```

Comment:
Oh I changed from `is` to `==` here by accident. I don't think `is` is proper though since it's not a singleton.

Reply:
Yeah, I feel like `==` is most often the right choice, so it's probably a good change.